MoreLinq.Test/TagFirstLastTest.cs

```diff
@@ -22,12 +22,29 @@ namespace MoreLinq.Test
     [TestFixture]
     public class TagFirstLastTest
     {
+        [Test]
+        public void TagFirstLastEvaluatesSourceLazily()
+        {
+            var source = MoreEnumerable.From(() => 123, () => 456, () => throw new TestException());
+            source.TagFirstLast((item, isFirst, isLast) => new { Item = item, IsFirst = isFirst, IsLast = isLast })
+                  .Take(1)
+                  .Consume();
+        }
+
         [Test]
         public void TagFirstLastIsLazy()
         {
             new BreakingSequence<object>().TagFirstLast(BreakingFunc.Of<object, bool, bool, object>());
         }
 
+        [Test]
+        public void TagFirstLastWithSourceSequenceOfZero()
+        {
+            var source = new int[0];
```

Comment:
```suggestion
            var source = EmptyArray<int>.Value;
```

Reply:
Sadly this doesn't compile when .NET Framework 4.5.1 is targeted.

Reply:
My bad, I didn't fully read your first comment.